docs/hello_nextflow/06_hello_config.md

```diff
@@ -1,247 +1,503 @@
+
+-   **`nextflow.config`** is a copy of the original `nextflow.config` file from the `hello-nextflow` directory, one level up (where we've been working so far).
+    Whenever there is a file named `nextflow.config` in the current directory, Nextflow will automatically load configuration from it. The one we have been using contains the following lines:
 
-1. Nextflow downloads the pipeline from the GitHub repository `seqeralabs/nf-hello-gatk`.
-1. It then executes the pipeline using the default configuration.
-1. The pipeline will likely use Docker containers to run the required tools (Samtools and GATK).
-1. It processes the input BAM files, creates index files, and performs variant calling.
-1. The results are generated by default in the `results` directory.
-1. Nextflow also creates a `work` directory containing intermediate files and logs.
-1. Upon completion, Nextflow displays a run summary, including any errors or warnings.
+    ```console title="nextflow.config" linenums="1"
+    docker.fixOwnership = true
+    docker.enabled = true
+    ```
 
-Now, let's see how this was configured and set up.
+    The `docker.fixOwnership = true` line is not really interesting.
+    It's a workaround for an issue that sometimes occur with containerized tools that set the wrong permissions on the files they write (which is the case with GenomicsDBImport in the GATK container image in our workflow).
 
-### 1.2. Check configuration
+    The `docker.enabled = true` line is what we care about here.
+    It specifies that Nextflow should use Docker to run process calls that specify a container image.
+    We're going to be playing with that shortly.
 
-Open the `nextflow.config` file and inspect the contents:
+!!!note
+
+    Anything you put into the `nextflow.config` can be overridden at runtime by providing the relevant process directives or parameters and values on the command line, or by importing another configuration file, according to the order of precedence described [here](https://www.nextflow.io/docs/latest/config.html).
+
+-   **`demo-params.json`** is a parameter file intended for supplying parameter values to a workflow.
+    We will use it in section 5 of this tutorial.
+
+The one thing that's missing is a way to point to the original data without making a copy of it or updating the file paths wherever they're specified.
+The simplest solution is to link to the data location.
+
+### 0.2. Create a symbolic link to the data
+
+Run this command from inside the `hello-config` directory:
 
 ```bash
-code nextflow.config
+ln -s ../data data
 ```
 
-The contents should look like this:
+This creates a symbolic link called `data` pointing to the data directory, which allows us to avoid having to change anything to how the file paths are set up.
 
-```groovy title="nextflow.config" linenums="1"
-docker.enabled = true
+```console title="Directory contents"
+hello-config
+├── data -> ../data
+├── demo-params.json
+├── solutions
```

Comment:
This output doesn't match what's in the directory. `solutions` is one level up, in `hello-nextflow`.

Reply:
has been updated